Assets/Scripts/UI/Overlay/OverlayMap.cs

```diff
@@ -351,11 +353,17 @@ private void Update()
             elapsed = 0f;
         }
 
+        if(currentOverlay != "None" && currentLayer != WorldController.Instance.cameraController.CurrentLayer)
+        {
+            Bake();
+            currentLayer = WorldController.Instance.cameraController.CurrentLayer;
+            elapsed = 0f;
+        }
         // TODO: Prettify.
```

Comment:
Well I pushed the change... somewhere. :p Github is a bit laggy laterly, let's see if it will update